pkg/storageincentives/agent.go

```diff
@@ -298,8 +320,25 @@ func (a *Agent) claim(ctx context.Context) error {
 	}
 
 	if isWinner {
+		a.mtx.Lock()
+		a.nodeStatus.State = winner.String()
```

Comment:
these "state" field is a bit ambiguous, it stores the current phase and other things like if the node won. 
Ideally there should be a `currentPhase` field, `isFrozen`, and `lastWonRound`

Reply:
According to the issue, the state should be either `claim | commit | reveal | idle | frozen`

We already have a `frozen` field in the state. So the `isFrozen` property might be redundant. Could you please  elaborate more on `currentPhase` and `lastWonRound` ?